File: dummy-ai-client/NetworkStandard/Messages/request_gameoperation_message.py
from NetworkStandard.MessageContainer.message_container import MessageContainer


class RequestGameOperationMessage(MessageContainer):

    def __init__(self, player_Id, type_enum, creation_date, debug_message,
                character_Id):
        super().__init__(player_Id, type_enum, creation_date, debug_message)
        self.character_Id = character_Id

File: backend/src/main/java/NetworkStandard/Messages/RequestGameOperationMessage.java
package NetworkStandard.Messages;

import NetworkStandard.MessageContainer.MessageContainer;
import NetworkStandard.MessageTypeEnum.MessageTypeEnum;

import java.util.Date;
import java.util.UUID;

/**
 * Message send by the server asks the client about the action, that the client wants to take with the character who is in charge.
 *
 * @author Standardisierungsdokument & Sedat Qaja
 */
public class RequestGameOperationMessage extends MessageContainer {
    UUID characterId;

    public RequestGameOperationMessage(UUID playerId, MessageTypeEnum type, Date creationDate, String debugMessage, UUID characterId) {
        super(playerId, MessageTypeEnum.REQUEST_GAME_OPERATION, creationDate, debugMessage);
        this.characterId = characterId;
    }
}
